dio/lib/src/dio_file_mode.dart

```diff
@@ -0,0 +1,24 @@
+/// {@template dio.DioFileMode}
+///  The file access mode when downloading the file.
+/// - [DioFileMode.write]: Mode for opening a file for reading and writing.
+///    The file is overwritten if it already exists. The file is created if it
+///    does not already exist
+/// - [DioFileMode.append]: Mode for opening a file for reading and writing
+///    to the end of it. The file is created if it does not already exist.
+/// {@endtemplate}
+enum DioFileMode {
```

Comment:
The name is a bit overwhelming. The type is only used for the download process.

Also the enhanced enum is not working on our minimum SDK support.

Reply:
I will convert it to extension

Reply:
any suggestion for naming